Update `@metamask/phishing-controller` to v2

Gudahtt opened the pull request:
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This update should dramatically reduce network traffic and double the frequency of phishing config updates.

This was accomplished by combining both of our phishing configurations into one list (the "stalelist"), then creating a separate list of the changes just the past few days (the "hotlist"). Now users will download a smaller list more frequently (every 30 minutes rather than every hour), whereas the full list is only updated every 4 days.

**Expected Behavior**

The phishing warning functionality should behave exactly as before, except that it will update slightly more frequently and use less network bandwidth. Previously updates would be downloaded between 0-90 minutes after the change was made, and now it's between 0-60 minutes instead.

We should do performance testing to ensure it doesn't negatively impact cold start as well. I don't anticipate it will, but it's touching the same area of the codebase as the previous cold start issue.

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

Gudahtt committed:
Update `@metamask/phishing-controller` to v2

This update should dramatically reduce network traffic and double the
frequency of phishing config updates.

This was accomplished by combining both of our phishing configurations
into one list (the "stalelist"), then creating a separate list of the
changes just the past few days (the "hotlist"). Now users will download
a smaller list more frequently (every 30 minutes rather than every
hour), whereas the full list is only updated every 4 days.

Gudahtt commented on patches/@metamask+phishing-controller+1.1.0.patch:
```diff
@@ -1,20 +0,0 @@
-             configs.push(metamaskConfig);
-         }
-+
-+        // Create a Set for more efficient checking when filtering phishfortHotlist.
```
The patched code here was replaced in the latest version. The equivalent to this code is now here: https://github.com/MetaMask/core/blob/main/packages/phishing-controller/src/utils.ts#L28-L40

It already uses `Set`s, and we did some quick benchmarking to ensure it was the fastest method of those we thought of.

Due to the way we've restructured the phishing updates, the list sizes being processed should be much smaller now as well.

Cal-L reviewed (approved):
LGTM! Before we merge - Could you add an expected behavior section? This would help QA know exactly what to look for when spot checking.

Gudahtt commented:
Sure, done!

Gudahtt commented:
Rebased to resolve a conflict.

I added the "needs-qa" label due to the performance concern, but feel free to substitute that for the "spot check" label if you think it's more appriopriate; I don't really know the criteria for when to use which.

Gudahtt commented on app/core/Engine.js:
```diff
@@ -253,6 +253,7 @@ class Engine {
       });
 
       const phishingController = new PhishingController();
+      phishingController.maybeUpdateState();
```
A call to `updatePhishingLists` was recently removed from this spot in #5797 to improve start time. I am unsure of the impact of that change; it might put the user at-risk of not being warned if the _very first page_ they visit is a phishing page, but I'm not sure and the risk is probably minimal.

An update call has been restored here for a few reasons:
* This `maybeUpdateState` call is conditional, and does nothing if the configuration is already sufficiently up-to-date
* Phishing config updates should be far less CPU-intensive with the new controller update
* This eliminates any risk to users for the first browser page load of that session.
